src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -1238,7 +1233,7 @@ private Action<Diagnostic> GetAddDiagnostic(ISymbol contextSymbol, ImmutableArra
         {
             return diagnostic =>
             {
-                if (isGeneratedCodeSymbol && shouldSuppressGeneratedCodeDiagnostic(diagnostic, analyzer, compilation, cancellationToken))
+                if (shouldSuppressGeneratedCodeDiagnostic(diagnostic, analyzer, compilation, cancellationToken))
```

Comment:
also, quite a bit of data in IsInGeneratedCode can be cached rather than repeatedly re-calculated for each diagnostics.

tree can be cached whether it is generated code or not. and if tree contains generated code, it looks like it can contain span for generated code part in the tree so that next time, it can just do contains to figure that information out.